apis/apps/v1alpha1/apicast_types.go

```diff
@@ -28,46 +28,81 @@ import (
+	// Kubernetes Service Account name to be used for the APIcast Deployment. The
+	// Service Account must exist beforehand.
 	// +optional
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+	// Image allows overriding the default APIcast gateway container image.
+	// This setting should only be used for dev/testing purposes. Setting
+	// this disables automated upgrades of the image.
 	// +optional
 	Image *string `json:"image,omitempty"`
+	// ExposedHost is the domain name used for external access. By default no
+	// external access is configured.
 	// +optional
 	ExposedHost *APIcastExposedHost `json:"exposedHost,omitempty"`
+	// DeploymentEnvironment is the environment for which the configuration will
+	// be downloaded from 3scale (Staging or Production), when using new APIcast.
+	// The value will also be used in the header X-3scale-User-Agent in the
+	// authorize/report requests made to 3scale Service Management API. It is
+	// used by 3scale just for statistics.
 	// +optional
 	DeploymentEnvironment *DeploymentEnvironmentType `json:"deploymentEnvironment,omitempty"` // THREESCALE_DEPLOYMENT_ENV
+	// DNSResolverAddress can be used to specify a custom DNS resolver address
+	// to be used by OpenResty.
 	// +optional
 	DNSResolverAddress *string `json:"dnsResolverAddress,omitempty"` // RESOLVER
+	// EnabledServices can be used to specify a list of service IDs used to
+	// filter the configured services.
 	// +optional
 	EnabledServices []string `json:"enabledServices,omitempty"` // APICAST_SERVICES_LIST
+	// ConfigurationLoadMode can be used to set APIcast's configuration load mode.
 	// +optional
 	// +kubebuilder:validation:Enum=boot;lazy
 	ConfigurationLoadMode *string `json:"configurationLoadMode,omitempty"` // APICAST_CONFIGURATION_LOADER
+	// LogLevel controls the log level of APIcast's OpenResty logs.
 	// +optional
 	// +kubebuilder:validation:Enum=debug;info;notice;warn;error;crit;alert;emerg
 	LogLevel *string `json:"logLevel,omitempty"` // APICAST_LOG_LEVEL
+	// PathRoutingEnabled can be used to enable APIcast's path-based routing
+	// in addition to to the default host-based routing.
 	// +optional
 	PathRoutingEnabled *bool `json:"pathRoutingEnabled,omitempty"` // APICAST_PATH_ROUTING
+	// ResponseCodesIncluded can be set to log the response codes of the responses
+	// returned by the API backend in 3scale.
 	// +optional
 	ResponseCodesIncluded *bool `json:"responseCodesIncluded,omitempty"` // APICAST_RESPONSE_CODES
+	// The period (in seconds) that the APIcast configuration will be stored in
+	// APIcast's cache.
 	// +optional
 	CacheConfigurationSeconds *int64 `json:"cacheConfigurationSeconds,omitempty"` // APICAST_CONFIGURATION_CACHE
+	// ManagementAPIScope controls APIcast Management API scope. The Management
+	// API is powerful and can control the APIcast configuration. debug level
+	// should only be enabled for debugging purposes.
```

Comment:
Should we list possible values?

Reply:
A little bit below you can find the values validation performet. Additionally, lots of the CR parameters belong to the APIcast realm  so user should already have this knowledge / consult it into the APIcast doc.